### PR TITLE
Stop using versioneer to manage versions and migrate wheel building for different CUDA versions

### PR DESCRIPTION
**Pull request by vyasr**
This change aligns ucx-py with the rest of RAPIDS plan for package and CUDA version management for conda and wheel builds going forward.

### CODE REVIEW

**Review by wence- — APPROVED** [review]
Thanks (you may wish to coordinate with @jakirkham's changes in #930)

**Review by ajschmidt8 — APPROVED** [review]

**Comment by vyasr** [comment]
Thanks for that pointer @wence- I'm discussing analogous changes to rmm with John at the moment so will coordinate on making this a consistent cross-repo effort.

**Comment by vyasr** [comment]
It looks like I don't have write permissions on the repo. @wence- when you get a chance would you mind merging this for me? Thanks!

File: pyproject.toml [hunk 1]
```diff
@@ -18,14 +18,11 @@ requires = [
...
     "tomli; python_version < '3.11'",
-    "versioneer>=0.24",
 ]
 
 [project]
 name = "ucx-py"
-dynamic = [
-    "version",
-]
+version = "0.31.0"
```
Comment (pentschev) [inline pyproject.toml:25]: Sorry, I was OOO and am only seeing this now. I'm wondering here if it's possible to programmatically get the version, as it was done before, maybe by looking at the branch name? If not, who's responsible for keeping this up-to-date, is it automatically managed by CI or do we have to do it manually?

File: ucp/__init__.py [hunk 2]
```diff
@@ -86,7 +85,7 @@ def _is_mig_device(handle):
     os.environ["UCX_MAX_RNDV_RAILS"] = "1"
 
 
-__version__ = _get_versions()["version"]
+__version__ = "0.31.0"
```
Comment (pentschev) [inline ucp/__init__.py:88]: Ditto.

**Review by pentschev — COMMENTED** [review]
@vyasr sorry for the delay but had a few questions here, could you take a look?

File: pyproject.toml (hunk 1 above)
Comment (jakirkham) [reply pyproject.toml:25]: Think the idea is that [the `update-version.sh` script]( https://github.com/rapidsai/ucx-py/blob/branch-0.31/ci/release/update-version.sh ) would handle this

File: ucp/__init__.py (hunk 2 above)
Comment (jakirkham) [reply ucp/__init__.py:88]: Same as above

File: pyproject.toml (hunk 1 above)
Comment (pentschev) [reply pyproject.toml:25]: Ah ok, got it now. Thanks John for clarifying!

Comment (vyasr) [reply pyproject.toml:25]: Yup that's right, thanks for answering here John. Long term I am hoping to do this even more programatically than we do now, but that has to do with improvements to update-version.sh and how the version there is produced.

**Pull request merged**